central/complianceoperator/v2/scanconfigurations/datastore/datastore_impl.go

```diff
@@ -186,3 +186,34 @@ func getScopeKeys(scanClusters []*storage.ComplianceOperatorScanConfigurationV2_
 
 	return clusterScopeKeys
 }
+
+func (ds *datastoreImpl) RemoveClusterFromScanConfig(ctx context.Context, clusterID string) error {
+	q := search.NewQueryBuilder().
+		AddExactMatches(search.ClusterID, clusterID).ProtoQuery()
+	scans, err := ds.GetScanConfigurations(ctx, q)
+	if err != nil {
+		return err
+	}
+	for _, scan := range scans {
+		clusters := scan.GetClusters()
+		newClusters := []*storage.ComplianceOperatorScanConfigurationV2_Cluster{}
+		for _, cluster := range clusters {
+			if cluster.GetClusterId() != clusterID {
+				newClusters = append(newClusters, cluster)
+			}
+
+		}
+		scan.Clusters = newClusters
+		err := ds.UpsertScanConfiguration(ctx, scan)
+		if err != nil {
+			return err
+		}
+		_, err = ds.statusStorage.DeleteByQuery(ctx, search.NewQueryBuilder().
+			AddExactMatches(search.ComplianceOperatorScanConfig, scan.GetId()).ProtoQuery())
```

Comment:
This means we're going to get rid of results from clusters that are removed, right?

Do we want to keep these cluster statuses around for historical purposes?

Reply:
We generally don't keep history around for objects that no longer exist (minus a few items like violations that follow a well defined retention period). Furthermore, it would be be an orphaned object with no way to garbage collect it.

Reply:
Actually this just gets rid of the scan configurations and related status.  It will not delete the results.  We should probably delete results too.  @ksurabhi91 can you write a ticket to delete the results as well.